tests/integration/mutation/update/field_kinds/one_to_many/with_alias_test.go

```diff
@@ -14,66 +14,108 @@ import (
+			testUtils.CollectionNamedMutationType,
+			testUtils.CollectionSaveMutationType,
+		}),
 		Actions: []any{
-			testUtils.Request{
-				Request: `mutation {
- 					create_Author(data: "{\"name\": \"John Grisham\"}") {
- 						_key
- 					}
- 				}`,
-				Results: []map[string]any{
-					{
-						"_key": author1Key,
-					},
-				},
+			testUtils.CreateDoc{
+				CollectionID: 1,
+				Doc: `{
+					"name": "John Grisham"
+				}`,
 			},
-			testUtils.Request{
-				Request: `mutation {
- 					create_Author(data: "{\"name\": \"New Shahzad\"}") {
- 						_key
- 					}
- 				}`,
-				Results: []map[string]any{
-					{
-						"_key": author2Key,
-					},
-				},
+			testUtils.CreateDoc{
+				CollectionID: 1,
+				Doc: `{
+					"name": "New Shahzad"
+				}`,
 			},
-			testUtils.Request{
-				Request: fmt.Sprintf(
-					`mutation {
- 						create_Book(data: "{\"name\": \"Painted House\",\"author\": \"%s\"}") {
- 							_key
- 							name
- 						}
- 					}`,
+			testUtils.CreateDoc{
+				CollectionID: 0,
+				Doc: fmt.Sprintf(
+					`{
+						"name": "Painted House",
+						"author": "%s"
+					}`,
 					author1Key,
 				),
-				Results: []map[string]any{
-					{
-						"_key": bookKey,
+			},
+			testUtils.UpdateDoc{
+				CollectionID: 1,
+				DocID:        1,
+				// NOTE: There is no `published` on book.
+				Doc: fmt.Sprintf(
+					`{
+						"published": "%s"
+					}`,
+					bookKey,
+				),
+				ExpectedError: "The given field does not exist. Name: published",
+			},
+		},
+	}
+
+	executeTestCase(t, test)
+}
+
+func TestMutationUpdateOneToMany_AliasRelationNameToLinkFromSingleSide_GQL(t *testing.T) {
+	author1Key := "bae-2edb7fdd-cad7-5ad4-9c7d-6920245a96ed"
+	bookKey := "bae-22e0a1c2-d12b-5bfd-b039-0cf72f963991"
+
+	test := testUtils.TestCase{
+		Description: "One to many update mutation using relation alias name from single side (wrong)",
+		// This restiction is temporary due to an inconsitent error message, see
+		// TestMutationUpdateOneToMany_AliasRelationNameToLinkFromSingleSide_Collection
+		// and https://github.com/sourcenetwork/defradb/issues/1854 for more info.
+		SupportedMutationTypes: immutable.Some([]testUtils.MutationType{
+			testUtils.GQLRequestMutationType,
+		}),
+		Actions: []any{
+			testUtils.CreateDoc{
+				CollectionID: 1,
+				Doc: `{
+					"name": "John Grisham"
+				}`,
+			},
+			testUtils.CreateDoc{
+				CollectionID: 1,
+				Doc: `{
+					"name": "New Shahzad"
+				}`,
+			},
+			testUtils.CreateDoc{
+				CollectionID: 0,
+				Doc: fmt.Sprintf(
+					`{
 						"name": "Painted House",
-					},
-				},
+						"author": "%s"
+					}`,
+					author1Key,
+				),
 			},
-			testUtils.Request{ // NOTE: There is no `published_id` and so `published` alias is invalid use on book.
-				Request: fmt.Sprintf(
-					`mutation {
- 						update_Author(id: "%s", data: "{\"published\": \"%s\"}") {
- 							name
- 						}
- 					}`,
-					author2Key,
+			testUtils.UpdateDoc{
+				CollectionID: 1,
+				DocID:        1,
```

Comment:
question: Should DocID be the `author2Key` here?

Reply:
no, it is and should be the (test) index of the document.